src/tasks/LibraryBuilder/Templates/autoinit.c

```diff
@@ -0,0 +1,104 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+
+#include <mono/jit/jit.h>
+#include <mono/jit/mono-private-unstable.h>
+#include <mono/metadata/assembly.h>
+
+#include "shared_library_log.h"
+
+static const char *bundle_path;
+
+void register_aot_modules (void);
+void preload_assemblies_with_exported_symbols ();
+typedef void (*MonoRuntimeInitCallback) (void);
+void mono_set_runtime_init_callback (MonoRuntimeInitCallback callback);
+
+static void
+cleanup_runtime_config (MonovmRuntimeConfigArguments *args, void *user_data)
+{
+    free ((void *)args->runtimeconfig.name.path);
+    free (args);
+    free (user_data);
+}
+
+static void
+initialize_runtimeconfig ()
+{
+    char *file_name = "runtimeconfig.bin";
+    size_t str_len = sizeof (char) * (strlen (bundle_path) + strlen (file_name) + 2); // +1 "/", +1 null-terminating char
+    char *file_path = (char *)malloc (str_len);
+    if (!file_path)
+        LOG_ERROR ("Out of memory.\n");
+
+    int num_char = snprintf (file_path, str_len, "%s/%s", bundle_path, file_name);
+    if (num_char < 0)
+        LOG_ERROR ("Encoding error while formatting '%s' and '%s' into \"%%s/%%s\".\n", bundle_path, file_name);
+
+    struct stat buffer;
+
+    if (stat (file_path, &buffer) == 0) {
+        MonovmRuntimeConfigArguments *arg = (MonovmRuntimeConfigArguments *)malloc (sizeof (MonovmRuntimeConfigArguments));
+        if (!arg)
+            LOG_ERROR ("Out of memory.\n");
+
+        arg->kind = 0;
+        arg->runtimeconfig.name.path = file_path;
+        monovm_runtimeconfig_initialize (arg, cleanup_runtime_config, NULL);
+    } else {
+        LOG_INFO ("Could not find file '%s'. Runtime configuration properties not initialized.\n", file_path);
```

Comment:
I think library init needs to be quite , if not having a runtimeconfig is a valid option that can happen, then we shouldn't log anything and if its not we should error out, but I think we should just accept that case and don't log anything.

Reply:
Removed LOG_INFO and removed from the header file.